src/status_im/chat/models/loading.cljs

```diff
@@ -69,15 +68,15 @@
 
 (rf/defn handle-mark-all-read-successful
   {:events [::mark-all-read-successful]}
-  [cofx]
-  (activity-center/notifications-fetch-unread-count cofx))
+  [_]
+  {:dispatch [:activity-center.notifications/fetch-unread-count]})
 
 (rf/defn handle-mark-all-read-in-community-successful
   {:events [::mark-all-read-in-community-successful]}
   [{:keys [db] :as cofx} chat-ids]
   (rf/merge cofx
-            {:db (reduce mark-chat-all-read db chat-ids)}
-            (activity-center/notifications-fetch-unread-count)))
+            {:db       (reduce mark-chat-all-read db chat-ids)
+             :dispatch [:activity-center.notifications/fetch-unread-count]}))
```

Comment:
yeah i see, here as well, do you know why we have a circular dependency, is it temporary while we're in moving process, is it possible to resolve now ?

Reply:
@flexsurfer, this is unfortunately a known downside of treating event handlers as functions as we do. It's very common to have events that need to dispatch to other events, and circular dependencies can easily happen. I had similar issues in past projects.

Similarly, when codebases use namespaced keywords, circular dependencies are also common, and that's one big reason why Clojure 1.11 supports the new `:as-alias` option, which only resolves keywords `::` for expansion, but doesn't actually load the namespace.

Reply:
yeah that's interesting, somehow we didn't have much  circular dependencies in the project before, i remember only a few cases in that case we have to move event to separate namespace

Reply:
> yeah that's interesting, somehow we didn't have much circular dependencies in the project before, i remember only a few cases in that case we have to move event to separate namespace

That's correct, most of the time we can rearrange the namespaces to accommodate for circular dependencies. It works well for the most part.